src/Uno.UWP/ApplicationModel/DataTransfer/DataPackage.macOS.cs

```diff
@@ -36,7 +42,32 @@ public partial class DataPackage
 		 *  WebLink                UTType.URL        NSPasteboard.NSPasteboardTypeUrl        NSPasteboard.NSUrlType 
 		 *  
 		 *  *1 : macOS 10.6 and later
-		 *  
+		 *
+		 * Below are more specific notes when converting types to/from macOS.
+		 *
+		 * --- UWP to macOS ---
+		 *
+		 * UWP has the following standard data formats that correspond with a macOS Url:
+		 *
+		 *  1. Uri, now deprecated in favor of:
+		 *  2. ApplicationLink and
+		 *  3. WebLink
+		 *
+		 * For maximum compatibility all are mapped to Url.
+		 * However, when applying data to the clipboard or drag/drop DataPackage, 
+		 * only one may be used at a time in the above defined priority. 
+		 * WebLink is considered more specific than ApplicationLink.
+		 *
+		 * --- macOS to UWP ---
+		 *
+		 * A macOS URL must be specially mapped for UWP as UWP's direct equivalent 
+		 * standard data format 'Uri' is deprecated.
+		 *
+		 * 1. WebLink is used if the macOS URL has a scheme of http or https 
+		 * 2. ApplicationLink is used if not #1
```

Comment:
As I was copying the code from the Android DataPackage updates I was also wondering how this could be put in one location to avoid duplication. Putting it as an extension or helper method some place would also allow us to quickly implement the logic on other platforms. Some ideas are below. Not sure where to locate these methods:

`bool IsWebLink(string url)` : would at least allow sharing the detection code, ideally we would like to share more logic.
`void SeparateUrl(string url, out string? webLink, out string? applicationLink)` : Slightly better as all logic could be shared now.

Going the other way:

`string CombineUrl(string? webLink, string? applicationLink, string? uri = null)`

If we get common methods some place I would be happy to update Skia. Generally speaking, I do have to be careful about spending much time on platforms that are not currently on the roadmap for my apps. As awesome as Skia is I must put that time elsewhere :)

Reply:
@dr1rrb When you have a moment please take a look at the changes in:

 * src/Uno.UWP/ApplicationModel/DataTransfer/DataPackage.cs and
 * src/Uno.UWP/ApplicationModel/DataTransfer/Clipboard.Android.cs

Corresponding with the last two commits. I went ahead and separate out this logic and added it to the Android clipboard. I think it does nicely simplify some things. If you are good with it I'll apply these same changes to macOS/Skia.